src/audio/pipeline.c

```diff
@@ -972,10 +972,10 @@ static uint64_t pipeline_task(void *arg)
 			trace_pipe_error_with_ids(p, "pipeline_task(): xrun "
 						  "recover failed! pipeline "
 						  "will be stopped!");
-			return 0; /* failed - host will stop this pipeline */
+			goto end;
 		}
 	}
-
-	tracev_pipe("pipeline_task() sched");
-	return p->ipc_pipe.period;
+end:
+	tracev_pipe("pipeline_task() done");
+	return SOF_TASK_STATE_COMPLETED;
```

Comment:
It will totally break timer driven scheduling. Low Latency Scheduler uses return value to calculate when it should execute the task.

Reply:
@tlauda but this task is schedule as idle task:

`schedule_task(&p->pipe_task, 0, p->ipc_pipe.period,
  937  		      SOF_SCHEDULE_FLAG_IDLE);`
 therefore its return value is abandoned:

`task->func(task->data);`

isn't it?

Reply:
Look at lib/ll_schedule.c, line 194:
`reschedule_usecs = ll_task->func(ll_task->data);`

Reply:
Okay, I have updated `pipeline_task` to return according to selected scheduler.